chopper/lib/src/request.dart

```diff
@@ -100,17 +71,17 @@ class Request extends http.BaseRequest {
   /// If [url] starts with 'http://' or 'https://', baseUrl is ignored.
   @visibleForTesting
   static Uri buildUri(
-    String baseUrl,
-    String url,
+    Uri baseUrl,
+    Uri url,
     Map<String, dynamic> parameters, {
     bool useBrackets = false,
     bool includeNullQueryVars = false,
   }) {
     // If the request's url is already a fully qualified URL, we can use it
     // as-is and ignore the baseUrl.
-    final Uri uri = url.startsWith('http://') || url.startsWith('https://')
-        ? Uri.parse(url)
-        : Uri.parse('${baseUrl.strip('/')}/${url.leftStrip('/')}');
+    final Uri uri = url.isScheme('HTTP') || url.isScheme('HTTPS')
+        ? _mergeUri(url, baseUrl, mergePaths: false)
```

Comment:
I added this with the idea of making it possible to preserve query parameters of the `baseUrl` (in the `Uri`). But I'm wondering if that is ideal. 

This makes it impossible the remove or override the query parameters provided from `baseUrl`. Only in the client calls (`get`, `post` etc) it's possible to override the `baseUrl` and also override any query parameter. But the `baseUrl`.

I kinda liked the idea of preserving the information but I can also imaging a lot of annoyances when its not preferred. What do you think @techouse?

Reply:
Hmm, to me putting query parameters into the `baseUrl` is just a crazy idea and more or less a hack.

Maybe you should add a validator that validates the `baseUrl` and throws an `Exception` when it detects query parameters in it. Even a simple `String.contains('?')` might do the trick.

Reply:
I guess its also kinda hacky besides its cleaner to provide a interceptor to add default query parameter if that would be needed.